### ISSUE TITLE
Access violation on Windows on GHC 8.8.3, 8.8.4, 8.10.1 and 8.10.2

### ISSUE THREAD

**Issue by asr** [agda/agda#4543]
The workflow using Windows and stack [fails](https://github.com/agda/agda/runs/507890135) on GHC 8.8.3:

```
alex                > Building executable 'alex' for alex-3.2.5..
alex                > 
alex                > Access violation in generated code when writing 0x0
alex                > 
alex                >  Attempting to reconstruct a stack trace...
``` 

It seems a similar problem was reported in the GHC bug tracker: [Issue 17926](https://gitlab.haskell.org/ghc/ghc/issues/17926).

Versions used by the workflow:
* Current runner version: '2.168.0'
* Operating System
  Microsoft Windows Server 2019
  10.0.17763
  Datacenter
* Virtual Environment
  Environment: windows-2019
  Version: 20200319.1

**Comment by asr** [comment]
Now we are using GHC 8.10.1 in the test of Windows + stack. Since problem in the OP doens't happen with this version of GHC, I'm closing this issue.

**Comment by L-TChen** [comment]
The problem sadly persists even with GHC 8.10.1 on Windows + stack, see https://github.com/agda/agda/runs/587539489.

**Comment by L-TChen** [comment]
After trying GHC versions from 8.10.1 to 8.4.4, unknown optimization flag(s) might be involved, since turning off all optimization flags (`-O0`) on 8.6.5 (or 8.8.2) works. 

For example, compiling [using GHC 8.8.2 without `--fast`](https://github.com/agda/agda/runs/588355517?check_suite_focus=true) fails, but compiling [using GHC 8.8.2 with `--fast`](https://github.com/agda/agda/runs/589135446?check_suite_focus=true) works. Why?

**Comment by asr** [comment]
@L-TChen, could you mention the above information in [Issue 17926](https://gitlab.haskell.org/ghc/ghc/issues/17926), please.

**Comment by L-TChen** [comment]
Sorry, I removed the log you mentioned in the Issue 17926 by mistake. So, I am building another set of configurations {8.8.2, 8.10.1} x {with `--fast`, without `--fast`}...

**Comment by asr** [comment]
The GHC issue will fixed in GHC 8.10.2 (see  [Issue 17926](https://gitlab.haskell.org/ghc/ghc/issues/17926)).

**Comment by asr** [comment]
The GHC issue was fixed in GHC 8.8.4 (see  [Issue 17926](https://gitlab.haskell.org/ghc/ghc/issues/17926)).

**Comment by asr** [comment]
It seems I'm getting the same problem on GHC 8.8.4 (see https://gitlab.haskell.org/ghc/ghc/-/issues/17926#note_294289).

**Comment by L-TChen** [comment]
Urgh... it seems this issue will not be fixed soon. Shall we try to find out which flags actually trigger this bug? I do not have Windows installed on any machine, though.

**Comment by asr** [comment]
Note we have the problem only with Cabal.

**Comment by asr** [comment]
According to https://gitlab.haskell.org/ghc/ghc/-/issues/18634 we need to wait for GHC 8.10.3.

**Comment by L-TChen** [comment]
I closed this issue accidentally instead of #4513. So, I am reopening this one and closed the other.

**Comment by L-TChen** [comment]
The problem seems going away as it usually failed at compiling `VersionCommit`, see the [log](https://github.com/L-TChen/agda/runs/1660063111?check_suite_focus=true).

Yet, `ghc` 8.10.3 still fails at compiling `TypeChecking.Rules.Term` with a different error message
```
[341 of 394] Compiling Agda.TypeChecking.Rules.Term ( src\full\Agda\TypeChecking\Rules\Term.hs, D:\a\agda\agda\dist-newstyle\build\x86_64-windows\ghc-8.10.3\Agda-2.6.2\build\Agda\TypeChecking\Rules\Term.o )
C://ProgramData//chocolatey//lib//ghc.8.10.3//tools//ghc-8.10.3//mingw//bin/as.exe: out of memory allocating 4064 bytes
`gcc.exe' failed in phase `Assembler'. (Exit code: 1)
Warning: Attempt 1 failed. Reason: Child_process exited with error code 1
```
This might be caused by something else, since `ghc` 8.10.2 sometimes fails randomly on the GitHub-hosted runner with the option `--split-sections`. Anyway, `cabal` manages to compile Agda. I will submit another issue if it fails reliably.

@asr, should we add a clause in `Agda.cabal` to exclude 8.8.x and 8.10.1--8.10.2 on Windows?

**Comment by asr** [comment]
> @asr, should we add a clause in `Agda.cabal` to exclude 8.8.x and 8.10.1--8.10.2 on Windows?

No. If the user has a similar problem with `cabal-install` we can suggest: to use `stack` or to use GHC 8.10.3.

**Comment by asr** [comment]
It seems the problem was fixed in GHC 8.10.3.
